src/coreclr/jit/importer.cpp

```diff
@@ -1795,6 +1796,12 @@ GenTree* Compiler::impNormStructVal(GenTree*             structVal,
             alreadyNormalized = true;
             break;
 
+        case GT_CNS_VEC:
+        {
+            assert(varTypeIsSIMD(structVal) && (structVal->gtType == structType));
+            break;
```

Comment:
Does this mean we'll be creating `OBJ(ADDR(CNS_VEC))`?

I would hate to see this hacky IR shape spread, the `makeTemp` solution was better.

Reply:
It means we're doing the same thing for `CNS_VEC` as we would've done for `GT_SIMD` and `GT_HWINTRINSIC`.

Reply:
So -- it creates `OBJ(ADDR(CNS_VEC))`?

How does it work given the code in morph and rationalization compensating for that IR shape is not present?

Reply:
As far as I'm aware all the relevant paths handle `OperIsSimdOrHWIntrinsic || IsCnsVec` already. Much of it flowing more generally from `varTypeIsSIMD`.

Reply:
And also this morph code:
```c
                    if (argx->OperIsSimdOrHWintrinsic() || argx->IsCnsVec() ||
                        (argx->OperIs(GT_OBJ) && argx->AsObj()->gtOp1->OperIs(GT_ADDR) &&
                         argx->AsObj()->gtOp1->AsOp()->gtOp1->OperIsSimdOrHWintrinsic()))
```

Reply:
Ah, these are both places that were handled originally and then undone based on previous feedback since it was creating `OBJ(ADDR(LCL_VAR))` rather than tracking the type and is now back to needing it to exist again.

It should be fixed now

Reply:
Yup, looks good to me now.